### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by nwlandry**
Added unit tests for the following files:
* matrix.py (Improved coverage from 77% to 95%)
* hypergraphviews.py (Improved coverage from 17% to 96%)
* function.py (Improved coverage from 60% to 84%)

Overall coverage improved from 61% to 66%.

Tests before:
test_before

Tests after:
test_after

Also made small fixes to the documentation.

### CODE REVIEW

File: tests/linalg/test_matrix.py [hunk 1]
```diff
@@ -51,43 +52,55 @@ def test_incidence_matrix(edgelist1, edgelist3, edgelist4):
...
+    assert I4[node_dict1[8], edge_dict1[3]] == 1
+
+    I5, node_dict, edge_dict = xgi.incidence_matrix(
         H1, index=True, weight=lambda node, edge, H: 3
     )
     node_dict4 = {k: v for v, k in node_dict.items()}
     edge_dict4 = {k: v for v, k in edge_dict.items()}
-    assert I4[node_dict4[1], edge_dict4[0]] == 3
+    assert I5[node_dict4[1], edge_dict4[0]] == 3
+    assert np.unique(np.ravel(I5.todense())).tolist() == [0, 3]
```
Comment (maximelucas) [inline tests/linalg/test_matrix.py:75]: break into two lines for readability?

File: tests/classes/test_function.py [hunk 2]
```diff
@@ -86,9 +86,129 @@ def test_create_empty_copy(edgelist1):
...
+    assert E2["name"] == "test"
+    assert E2["timestamp"] == "Nov. 20"
+    with pytest.raises(XGIError):
+        author = E2["author"]
+    for n in E2.nodes:
+        assert E2.nodes[n]["name"] == attr_dict[n]["name"]
+
+
+def test_set_node_attributes(edgelist1):
+    attr_dict1 = {
```
Comment (maximelucas) [inline tests/classes/test_function.py:98]: add test for what happens if trying to set attr for a node not in the hypergraph?
attr_dict = {0 : {"name" : "Nobody"}}
From the function itself it looks like no error is raised, should we raise one? (Same for the set_edge_attr)

File: tests/classes/test_function.py [hunk 3]
```diff
@@ -86,9 +86,129 @@ def test_create_empty_copy(edgelist1):
...
+        3: {"name": "Alice"},
+        4: {"name": "Giovanni"},
+        5: {"name": "Heather"},
+        6: {"name": "Juan"},
+        7: {"name": "Nicole"},
+        8: {"name": "Sinan"},
+    }
+    xgi.set_node_attributes(H1, attr_dict)
+
+    assert xgi.get_node_attributes(H1, "name") == {id : data["name"] for id, data in attr_dict.items()}
```
Comment (maximelucas) [inline tests/classes/test_function.py]: This still works if we set the name of a subset of the nodes I think, right?
LIke attr_dict = { 1: {"name": "Leonie"}}

File: tests/classes/test_function.py [hunk 4]
```diff
@@ -86,9 +86,129 @@ def test_create_empty_copy(edgelist1):
...
+    assert xgi.get_node_attributes(H1, "weight") == dict()
+    
+
+def test_set_edge_attributes(edgelist1):
+    H1 = xgi.Hypergraph(edgelist1)
+    attr_dict1 = {
+        0: {"weight": 1},
+        1: {"weight": 2},
+        2: {"weight": 3},
```
[the remark this answers is not in the document]
Comment (nwlandry) [reply tests/classes/test_function.py:172]: Done.

File: tests/linalg/test_matrix.py (hunk 1 above)
Comment (nwlandry) [reply tests/linalg/test_matrix.py:75]: Done.

File: tests/classes/test_function.py (hunk 2 above)
Comment (nwlandry) [reply tests/classes/test_function.py:98]: I'm torn on this, but my inclination is that it's okay if it fails silently. Like if you had a giant list of attribute data some of which corresponded to nodes in the hypergraph and others which didn't (For instance the Project 90 network data set), you would potentially have to remove the attributes connected to non-existent nodes for it to run.

Comment (maximelucas) [reply tests/classes/test_function.py:98]: Okay, fair enough.

**Review by maximelucas — APPROVED** [review]

**Pull request merged**